tests/river/e2e/api/test_preview.py

```diff
@@ -1,36 +1,96 @@
+from unittest.mock import patch
+
 import pytest
 
 from django.urls import reverse
 
 from syrupy.filters import paths
 
-pytestmark = pytest.mark.django_db
+pytestmark = [pytest.mark.django_db, pytest.mark.fhir_api]
 
 
 @pytest.mark.export_data("mimic_mapping.json")
-def test_preview(api_client, export_data, snapshot):
-    url = reverse("sources-list")
+class TestPreview:
+    @pytest.fixture
+    def patient_mapping(self, api_client, export_data):
+        url = reverse("sources-list")
+
+        response = api_client.post(url + "import/", export_data, format="json")
+
+        assert response.status_code == 201, response.data
+
+        patient_mapping_id = next(
+            r["id"] for r in response.json()["resources"] if r["label"].startswith("feat_6_join")
+        )
+        return patient_mapping_id
+
+    def test_preview(self, api_client, patient_mapping, snapshot):
+        url = reverse("preview")
+
+        data = {
+            "resource_id": patient_mapping,
+            "primary_key_values": ["10006"],
+        }
+        response = api_client.post(url, data, format="json")
+
+        assert response.status_code == 200, response.data
+        assert len(response.data["errors"]) == 0, response.data["errors"]
+        assert len(response.data["instances"]) == 1
+        fhir_instance = response.data["instances"][0]
+
+        # exclude the following attributes from the snapshort assertion:
+        # - lastUpdated: changes at each run
+        # - meta.tag.*.code: source_id or resource_id,
+        # changes at each run since the mapping is re-imported
+        assert fhir_instance == snapshot(exclude=paths("meta.lastUpdated", "meta.tag.0.code", "meta.tag.1.code"))
+
+    def test_preview_not_found(self, api_client, patient_mapping, snapshot):
+        url = reverse("preview")
+
+        data = {
+            "resource_id": patient_mapping,
+            "primary_key_values": ["123456789"],  # does not exist
+        }
+        response = api_client.post(url, data, format="json")
+
+        assert response.status_code == 200, response.data
+        assert len(response.data["errors"]) == 0, response.data["errors"]
+        assert len(response.data["instances"]) == 0
 
-    response = api_client.post(url + "import/", export_data, format="json")
+    @patch(
```

Comment:
Why not using the in-memory adapter?

Reply:
I am using the in-memory adapter:
```
# django/settings/test.py
DEFAULT_FHIR_API_CLASS = "common.adapters.fhir_api.InMemoryFhirAPI"
```
 I don't like this mock either but I can't think of a better way of customizing the behaviour of the class. We discussed it briefly with teva and agreed on the fact that we needed to refactor this.

Reply:
Maybe some kind of mechanism to make the fake api return an error on validate (toggle an attribute, add a dummy field in the input, etc.)?

Reply:
as discussed, we need to refactor